geode-core/src/main/java/org/apache/geode/internal/cache/TXEntryState.java

```diff
@@ -238,6 +238,7 @@
    */
   private static final boolean VERBOSE_CONFLICT_STRING =
       Boolean.getBoolean(DistributionConfig.GEMFIRE_PREFIX + "verboseConflictString");
+  private boolean detectReadConflicts;
```

Comment:
This field should be on TXRegionState instead of TXEntryState.
Each TXEntryState knows the TXRegionState it belong to (you can call getTXRegionState()) so in the isDirty method you can ask your TXRegionState if it detects read conflicts.

If this field can not change make it "final".